androgee/androgee.py

```diff
@@ -1,74 +1,51 @@
-import os
-import sys
 import random
 import logging
 import discord
-from discord.ext import commands
+from discord.ext import typed_commands as commands
 from pathlib import (
     Path,
-)  # This makes working with files and searching through them a piece of cake
+)
+from androgee.cogs.spam_badwords import BadWords
+from androgee.init import COMMAND_PREFIX, BOT_TOKEN, MOD_ROLE_ID
 
-# I think it'd be better if we check if environment
-# variables are present before doing anything else.
-try:
-    mod_role_id = os.environ["mod_role_id"]
-    mod_role_name = os.environ["mod_role_name"]
-    COMMAND_PREFIX = os.environ["DISCORD_PREFIX"]
-    BOT_TOKEN = os.environ["DISCORD_TOKEN"]
-except KeyError as e:
-    print(f"The {e} environment variable is missing! Androgee cannot run!")
-    sys.exit(1)
 
 logging.basicConfig(level=logging.WARNING)
-bot = commands.Bot(command_prefix=COMMAND_PREFIX)
+bot: commands.Bot = commands.Bot(command_prefix=COMMAND_PREFIX)
 media_folder = Path(
     __file__, "../media"
 ).resolve()  # Root of the media folder from file.
 
 
-@bot.event
-async def on_ready():
-    print(f"We is logged in as {bot.user}")
+# Super quick and easy random selection, partial credit to citrusMarmelade
+def get_image(ctx):
+    images = list(media_folder.glob(f"{ctx.command}/*"))
+    return discord.File(random.choice(images))
 
 
 @bot.command(name="spray", aliases=["spritzered"])
-@commands.has_any_role(mod_role_name, mod_role_id)
 async def spray(ctx, member: discord.Member = None):
-    try:
-        image = get_image(ctx)
-        if member is None:
-            await ctx.send(file=image)
-        else:
-            ctx.send(f"!tempmute {member.mention} 300 being too rowdy")
-            message = f"{member.mention} was sprirzered by {ctx.message.author.mention}"
-            await ctx.send(message, file=image)
-    except commands.MissingRole:
-        image = get_image(ctx)
-        if member is None:
-            await ctx.send(file=image)
-        else:
-            message = f"{member.mention} was sprirzered by {ctx.message.author.mention}"
-            await ctx.send(message, file=image)
+    if MOD_ROLE_ID in [y.id for y in ctx.author.roles] and member is not None:
```

Comment:
I'm not sure the -spray command needs to say anything. I thought it was just a user-command for people (heroes?) to spray other people.

That rework could be for a different PR though. tl;dr I'm not sure lines 27, 28, and 29 are needed.

Reply:
the tldr of this change is that I felt it would be nice for every one to be able to use the -spray and -bonk command but only the admins to have a effect